### PR TITLE
Use strategies validations for proposal validation

### PR DESCRIPTION
**Pull request by samuveth**
Fixes part #3466

There is lots of refactoring, particularly noteworthy is the `MessageWarningValidation.vue` component, which now functions as the component for both voting and proposal validation messages.

@j0hnfl0w, even though you may not have full context. If you could, please take a moment to skim over the code and comment on anything that catches your eye.

### CODE REVIEW

File: src/components/ModalVote.vue [hunk 1]
```diff
@@ -91,8 +91,10 @@ async function handleSubmit() {
 }
 
 async function loadVotingValidation() {
-  if (!props.proposal.validation || props.proposal.validation.name === 'any')
-    return (isValidVoter.value = true);
+  if (!props.proposal.validation || props.proposal.validation.name === 'any') {
+    isValidVoter.value = true;
+    return;
+  }
```
Comment (ChaituVR) [inline src/components/ModalVote.vue:97]: This is nice, is it possible to similar check for proposal validations too? can't find it 

Check if it is basic validation and minScore is 0

File: src/helpers/snapshot.ts [hunk 2]
```diff
@@ -117,10 +120,15 @@ export async function proposalValidation(
     space.id,
     space.network,
     'latest',
-    // this is needed until we change proposal validation
-    { minScore: space.filters.minScore, strategies: space.strategies },
+    {
+      minScore: space.validation?.params?.minScore || space.filters.minScore,
+      strategies: space.validation?.params?.strategies || space.strategies
+    },
```
Comment (ChaituVR) [inline src/helpers/snapshot.ts:126]: I just noticed we are not passing other params here, maybe should be like
```suggestion
    {
      minScore: space.validation?.params?.minScore || space.filters.minScore,
      strategies: space.validation?.params?.strategies || space.strategies
      ...space.validation?.params
    },
```

But also we don't use minScore and strategies on passport validation, may these should be optional, not sure how 😄

File: src/locales/default.json [hunk 3]
```diff
@@ -610,20 +600,50 @@
...
+      "description": "Anyone can create a proposal."
+    },
+    "basic": {
+      "label": "Basic",
+      "description": "Use any strategy to determine if a user can create a proposal.",
+      "invalidVoterMessage": "You do not meet the minimum balance requirement of {amount} {symbol} to create a proposal."
+    },
+    "passport-gated": {
+      "label": "Gitcoin Passport gated",
+      "description": "Protect your space from spam by requiring users to have a Gitcoin Passport to create a proposal.",
```
Comment (ChaituVR) [inline src/locales/default.json:624]: Is there some way to make these optional. and if not configured, we read them from https://score.snapshot.org/api/validations

else we may have to ask everyone that create a validation to add a title and description here

File: src/helpers/snapshot.ts (hunk 2 above)
Comment (ChaituVR) [reply src/helpers/snapshot.ts:126]: Maybe we can pass minScore and strategies only for basic validation?

Comment (ChaituVR) [reply src/helpers/snapshot.ts:126]: I mean like here https://github.com/snapshot-labs/snapshot-sequencer/pull/30/files (Didn't test it)

File: src/components/ModalVote.vue (hunk 1 above)
Comment (samuveth) [reply src/components/ModalVote.vue:97]: Yes good catch, added that :)

File: src/helpers/snapshot.ts (hunk 2 above)
Comment (samuveth) [reply src/helpers/snapshot.ts:126]: Yes, good catch again!

Comment (samuveth) [reply src/helpers/snapshot.ts]: How is [this](https://github.com/snapshot-labs/snapshot/pull/3558/commits/6192eff5bdb88364f538ac2ffaf5b6dc18874e6e)?

File: src/locales/default.json (hunk 3 above)
Comment (samuveth) [reply src/locales/default.json:624]: You can add/keep title and description for validations on strategies repo if you like, just on our UI I haven't found a good way to use them everywhere we need them. So for now we will have it like this

**Comment by samuveth** [comment]
@ChaituVR I think we can merge this regardless of the sequencer PR, wdyt?

File: src/views/SpaceCreate.vue [hunk 4]
```diff
@@ -263,6 +263,11 @@ async function validateAuthor() {
       return;
     }
 
+    if (props.space.validation.name === 'any') {
+      isValidAuthor.value = true;
+      return;
+    }
```
Comment (ChaituVR) [inline src/views/SpaceCreate.vue:269]: It still goes through if a space has `basic` and `minScore` considering most spaces have this right now. 
```json
"validation": {
"name": "basic",
"params": {}
}
```

Should be fine too, will just reduce few calls to score API ;)

Comment (samuveth) [reply src/views/SpaceCreate.vue:269]: Yep good point will change that

**Review by ChaituVR — APPROVED** [review]
Everything else is working well and great as far as i tested.

TestedAck

**Pull request merged**